core/core.go

```diff
@@ -119,6 +120,12 @@ func New(config Config) (*App, error) {
 		return nil, err
 	}
 
+	// Check if the DB has been previously intialized with a different networkId
+	err = initNetworkId(config.EthereumNetworkID, meshDB)
```

Comment:
For cases like this, when a function returns an `error` as it's only return value, we like to use the following shorthand:

```go
if err := initNetworkId(config.EthereumNetworkID, meshDB); err != nil {
    return nil, err
}
```

It has the advantage of limiting the scope of `err`.